numpy/distutils/misc_util.py

```diff
@@ -14,9 +14,15 @@
 import distutils
 from distutils.errors import DistutilsError
 try:
-    from threading import local as tlocal
+    from threading import local as tlocal, Lock
 except ImportError:
-    from dummy_threading import local as tlocal
+    from dummy_threading import local as tlocal, Lock
+
```

Comment:
I assume module imports can only happen once, if not there is nothing we could do as we can't initialize a lock

registering cleanup could be done, though it only saves one line. maybe also the lock if we assume append is threadsafe (which it probably is in cpython)